MedallionShell/AttachedCommand.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Medallion.Shell.Streams;
+
+namespace Medallion.Shell
+{
+    internal sealed class AttachedCommand : Command
+    {
+        private const string StreamPropertyExceptionMessage =
+            "This property cannot be used when attaching to already running process.";
+        private readonly Process process;
+        private readonly Task<CommandResult> commandResultTask;
+        
+        internal AttachedCommand(
+            Process process,
+            bool throwOnError,
+            TimeSpan timeout,
+            CancellationToken cancellationToken)
+        {
+            this.process = process;
+            var processMonitoringTask = System.Threading.Tasks.Task.Run(() => process.WaitForExit());
```

Comment:
This is simple, but it is also less efficient thread-wise than what ProcessCommand does using events. Of course, using events is trickier here because we have 2 race conditions to deal with:

(1) if the process exits before/during the call to EnableRaisingEvents, then EnableRaisingEvents throws an InvalidOperationException.

(2) If the process exits between EnableRaisingEvents and our setting the event, then we never see the exit event.

That said, we can work around both of these:

```
private static Task CreateProcessMonitoringTask(Process process)
{
    var taskBuilder = new TaskCompletionSource<bool>();

    // EnableRaisingEvents will throw if the process has already exited; to account for
    // that race condition we return a simple blocking task in that case
    try { process.EnableRaisingEvents = true; }
    catch { return Task.Run() => process.WaitForExit()); }

    process.Exited += (o, e) => taskBuilder.TrySetResult(false);

    // we must account for the race condition where the process exits between enabling events and
    // subscribing to Exited. Therefore, we do a non-blocking exit check after the subscription to account
    // for this
    if (process.WaitForExit(0)) { taskBuilder.TrySetResult(false); }
  
    return taskBuilder.Task;
}
```

Reply:
I was thinking about doing something like that, but eventually returned to the simple version. One question though: why not use `HasExited` instead of `WaitForExit(0)`?